data/campaigns/tutorial/scenarios/03_Tutorial_part_3.cfg

```diff
@@ -0,0 +1,803 @@
+            [/have_unit]
+            [then]
+                [message]
+                    speaker=narrator
+                    caption= _ "Recalling"
+                    image=wesnoth-icon.png
+                    message= _ "In campaigns, you can normally <i>recall</i> your experienced veterans from previous scenarios. It’s unusual for a campaign to switch to the opposite sides, but in this scenario the tutorial does, and for ease of balancing the difficulty your recall list isn’t carried to this scenario. However, you start with a level 2 Orcish Ruler, whose leadership works exactly as the Elvish Captain’s did."
+                [/message]
+            [/then]
+            [else]
+                [message]
+                    speaker=narrator
+                    caption= _ "Recalling"
+                    image=wesnoth-icon.png
+                    message= _ "In campaigns, you can normally <i>recall</i> your experienced veterans from previous scenarios. It’s unusual for a campaign to switch to the opposite sides, but in this scenario the tutorial does, and for ease of balancing the difficulty your recall list isn’t carried to this scenario. However, you start with a level 2 Orcish Warrior."
+                [/message]
+            [/else]
+        [/if]
+
+        {GENDER (
+            [message]
+                speaker=Grudush
+                message= _ "You can recruit the types of orcish units that Thrag recruited in the previous scenario (Grunts, Archers and Wolf Riders). You also have a new unit type available, the Orcish Assassin."
+            [/message]
+        ) (
+            [message]
+                speaker=Grudush
+                # po: the male/female on this message is for the "You"
+                message= _ "female^You can recruit the types of orcish units that Thrag recruited in the previous scenario (Grunts, Archers and Wolf Riders). You also have a new unit type available, the Orcish Assassin."
+            [/message]
+        )}
+    [/event]
+
+    [event]
+        name=exit_hex
+        [filter]
+            id=Dumbo
+        [/filter]
+        first_time_only=yes
+        [filter_condition]
+            [variable]
+                name=spoke_about_dumbo_trap
+                boolean_equals=no
+            [/variable]
+        [/filter_condition]
+
+        [if]
+            {VARIABLE_CONDITIONAL y2 less_than $y1}
+            [then]
+                # The scout was about to be moved north, towards the trap. This also triggers if the player moves him into the water.
+                {TALK_ABOUT Dumbo ( _ "Speaking of not being too brash, let’s discuss the trap north of our scout before moving him.")}
+            [/then]
+            [else]
+                {TALK_ABOUT Dumbo ( _ "Good idea, moving the scout away from the elves. But let’s discuss the trap north of him before moving him.")}
+            [/else]
+        [/if]
+
+        [cancel_action][/cancel_action]
+
+        [fire_event]
```

Comment:
I've redesigned this, it now uses a ghost-like copy of the unit, and only interrupts the player if they move to the north end of the bridge. Otherwise the ZoC discussion happens at the end of the turn.